### PR TITLE
Add `Set` support for `hasAny`/`hasAll` expression

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do

# Motivation

# Additional Notes

Jira ticket: [DEBUG-2488]

[DEBUG-2488]: https://datadoghq.atlassian.net/browse/DEBUG-2488?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-java-agent/agent-debugger/debugger-el/src/main/java/com/datadog/debugger/el/expressions/HasAllExpression.java [hunk 1]
```diff
@@ -69,11 +73,30 @@ public Boolean evaluate(ValueReferenceResolver valueRefResolver) {
...
       return Boolean.TRUE;
-    } else {
-      return filterPredicateExpression.evaluate(
-          valueRefResolver.withExtensions(
-              Collections.singletonMap(ValueReferences.ITERATOR_EXTENSION_NAME, value)));
     }
+    if (value instanceof SetValue) {
+      SetValue set = (SetValue) value;
+      if (set.isEmpty()) {
+        return Boolean.FALSE;
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/debugger-el/src/main/java/com/datadog/debugger/el/expressions/HasAllExpression.java:80]: is that constant between implementations?

in javascript (every(lambda)) and C# (Any(lambda)) they return TRUE for if collection is empty.

File: dd-java-agent/agent-debugger/debugger-el/src/main/java/com/datadog/debugger/el/expressions/HasAnyExpression.java [hunk 2]
```diff
@@ -70,11 +74,31 @@ public Boolean evaluate(ValueReferenceResolver valueRefResolver) {
...
       return Boolean.FALSE;
-    } else {
-      return filterPredicateExpression.evaluate(
-          valueRefResolver.withExtensions(
-              Collections.singletonMap(ValueReferences.ITERATOR_EXTENSION_NAME, value)));
     }
+    if (value instanceof SetValue) {
+      SetValue set = (SetValue) value;
+      if (set.isEmpty()) {
+        return Boolean.FALSE;
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/debugger-el/src/main/java/com/datadog/debugger/el/expressions/HasAnyExpression.java:81]: other languages return FALSE as well, so this one is standard.

File: dd-java-agent/agent-debugger/debugger-el/src/test/java/com/datadog/debugger/el/expressions/HasAllExpressionTest.java [hunk 3]
```diff
@@ -167,7 +169,7 @@ void testListHasAll() {
   }
 
   @Test
-  void testMapHasAny() {
+  void testMapHasAll() {
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/debugger-el/src/test/java/com/datadog/debugger/el/expressions/HasAllExpressionTest.java:172]: no test for empty collection?

**Review by shatzi — APPROVED** [review]
we should have system-tests and update the EL RFC to specify what this method should do in case of null + empty collections.

File: dd-java-agent/agent-debugger/debugger-el/src/main/java/com/datadog/debugger/el/expressions/HasAllExpression.java (hunk 1 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/debugger-el/src/main/java/com/datadog/debugger/el/expressions/HasAllExpression.java]: fixed

File: dd-java-agent/agent-debugger/debugger-el/src/test/java/com/datadog/debugger/el/expressions/HasAllExpressionTest.java (hunk 3 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/debugger-el/src/test/java/com/datadog/debugger/el/expressions/HasAllExpressionTest.java:172]: added

**Pull request merged**